### PR TITLE
Extract Xxd class for iOS AOT snapshotting

### PR DESCRIPTION
**Pull request by cbracken**
Allows for better mockability when tests are added.

### CODE REVIEW

File: packages/flutter_tools/lib/src/ios/mac.dart
Comment (goderbauer) [inline packages/flutter_tools/lib/src/ios/mac.dart]: This seems unused?

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_tools/lib/src/ios/mac.dart
Comment (cbracken) [reply packages/flutter_tools/lib/src/ios/mac.dart]: Agh - nice catch. Yeah it's in #16974 which has been waiting for tip-of-tree to go green before I can land it.

**Review by goderbauer — APPROVED** [review]
LGTM

**Comment by tvolkert** [comment]
This broke several tests - reverting.

**Pull request merged**
